### ISSUE TITLE
Numeric field support

### ISSUE THREAD

**Issue by guanlisheng** [moneymanagerex/mmex-ios#2]
As statement in #1 that [SQLite.swift](https://github.com/stephencelis/SQLite.swift) has some limitation to support [Numeric type](https://www.sqlite.org/datatype3.html)

**Comment by guanlisheng** [comment]
here you go a workaround by declaring two Expressions

```
extension Row {
    func getNumeric(_ doubleExpression: Expression<Double?>, _ intExpression: Expression<Int64?>) -> Double {
        return self[doubleExpression] ?? Double(self[intExpression] ?? 0)
    }
}
```

**Comment by georgeef** [comment]
Something like this could be a workaround at the level of Swift library. However, the proper fix is to use the sqlite3_column_* functions, which perform the type conversions at the level of C library.

Type conversions at Swift level have a small risk of incompatibility with respect to the C library, as used by other languages (and probably by MMEX Desktop through the C++ library). Better to avoid them. SQLite.swift should be only syntactic sugar, it should not process data behind the scenes by itself.

**Comment by georgeef** [comment]
A fix for NUMERIC fields has been implemented in #1.

In the same PR there is also some refactoring and a few more fixes for Current, Account, Category, Payee. Another PR will refactor Transaction similarly.

**Comment by georgeef** [comment]
At the moment there are two workarounds which do not require any patch of SQLite.swift
- Use `cast()` on Expressions in order to convert NUMERIC to REAL inside libsqlite3.
- Try to read `Double?`, and if it returns nil try to read an integer which is then converted to Double.
